pkg/proxy/winkernel/hns_test.go

```diff
@@ -0,0 +1,488 @@
+// +build wintest
+
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package winkernel
+
+import (
+	"github.com/Microsoft/hcsshim"
+	"net"
+	"testing"
+)
+
+const sourceVip = "192.168.1.2"
+const serviceVip = "11.0.0.1"
+const addressPrefix = "192.168.1.0/24"
+const gatewayAddress = "192.168.1.1"
+const epMacAddress = "00-11-22-33-44-55"
+const epIpAddress = "192.168.1.3"
+const protocol = 6
+const internalPort = 80
+const externalPort = 32440
+
+func CreateTestNetwork() (*hcsshim.HNSNetwork, error) {
+	network := &hcsshim.HNSNetwork{
+
+		Type: "NAT",
+
+		Name: "testnat",
+
+		Subnets: []hcsshim.Subnet{
+
+			{
+
+				AddressPrefix:  "192.168.1.0/24",
+
+				GatewayAddress: "192.168.1.1",
+
+			},
+
+		},
+
+	}
+	return network.Create()
+
+}
+
+func TestGetNetworkByName(t *testing.T) {
+	hnsV1 := hnsV1{}
```

Comment:
Test code could also work on the interface object, rather than trying out methods in each interface. If a new interface impl comes up tomorrow, I don't have to change this test code, if it is interface based.

Reply:
I think I still need to make sure my implementations are correct, right? With these tests, I've already found several bugs

Reply:
Yes, test are required, but just asking to make it interface based like kubeproxy code. See if you can pass in interface via param or a global param etc. Test doesn't have to know what interface it is using & you can run the test using v1 separately & v2 separately to validate your impl